WMFData/Sources/WMFData/Data Controllers/Activity Tab/WMFActivityTabDataController.swift

```diff
@@ -214,22 +214,63 @@ public actor WMFActivityTabDataController {
     }
 
     public func getTimelineItems() async throws -> [Date: [TimelineItem]] {
-        var allItems: [Date: [TimelineItem]] = [:]
-
-        let savedItems = try await fetchTimelineSavedArticles()
+        let rawSavedItems = try await fetchTimelineSavedArticles()
         let readItems = try await fetchTimelineReadArticles()
 
-        allItems.merge(savedItems) { old, new in
-            return old + new
+        let dedupedSavedItems = Self.deduplicatedSavedItems(rawSavedItems)
+
+        var allItems: [Date: [TimelineItem]] = [:]
+
+        allItems.merge(dedupedSavedItems) { old, new in
+            old + new
         }
 
         allItems.merge(readItems) { old, new in
-            return old + new
+            old + new
         }
 
         return allItems
     }
 
+    private static func deduplicatedSavedItems(_ savedItems: [Date: [TimelineItem]]) -> [Date: [TimelineItem]] {
```

Comment:
Note: This function might be overkill, but I had trouble reproducing the issue (maybe due to constant re-installs for development)